### PR TITLE
Refactor to support multiple hop paths

### PR DESCRIPTION
**Pull request by 0xh3rman**
Fixes https://github.com/gemwalletcom/core/issues/320 by introducing 2 hops swap support.

Changes: 
- Add base pair config for each chain (WETH / dominant stable coin)
- Check 2 hops quote for pari not against base pairs (same fee_tier)
- - token in > weth > token out
- - token in > usdc > token out

Quote Test
- [x] Optimism
- [x] Ethereum
- [x] Smart Chain

Swap Test
- [x] Op
- [x] Smart Chain

### CODE REVIEW

File: crates/gem_evm/src/uniswap/path.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+impl From<AssetId> for EthereumAddress {
+    fn from(asset_id: AssetId) -> Self {
+        let token_id = asset_id.token_id.unwrap_or_default();
+        EthereumAddress::parse(&token_id).unwrap()
+    }
+}
+
+pub fn get_base_pair(chain: &EVMChain) -> BasePair {
+    let weth = chain.weth_contract().unwrap().to_string();
```
Comment (gemcoder21) [inline crates/gem_evm/src/uniswap/path.rs:52]: avoid unwrap? 

can we support multiple stable coins per chain?

**Review by gemcoder21 — APPROVED** [review]

File: crates/gem_evm/src/uniswap/path.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+pub fn build_direct_pair(token_in: &EthereumAddress, token_out: &EthereumAddress, fee_tier: u32) -> Bytes {
+    let mut bytes: Vec<u8> = vec![];
+    let fee = U24::from(fee_tier);
+    bytes.extend(&token_in.bytes);
+    bytes.extend(&fee.to_be_bytes_vec());
+    bytes.extend(&token_out.bytes);
+    Bytes::from(bytes)
+}
+
+pub fn validate_pairs(token_pairs: &[TokenPair]) -> bool {
```
Comment (gemcoder21) [inline crates/gem_evm/src/uniswap/path.rs:82]:
```
pub fn validate_pairs(token_pairs: &[TokenPair]) -> bool {
    token_pairs
        .windows(2)
        .all(|pair| pair[0].token_out == pair[1].token_in)
}
```

does it do the same?

File: crates/gem_evm/src/uniswap/path.rs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+impl From<AssetId> for EthereumAddress {
+    fn from(asset_id: AssetId) -> Self {
+        let token_id = asset_id.token_id.unwrap_or_default();
+        EthereumAddress::parse(&token_id).unwrap()
+    }
+}
+
+pub fn get_base_pair(chain: &EVMChain) -> Option<BasePair> {
```
Comment (gemcoder21) [inline crates/gem_evm/src/uniswap/path.rs:51]: let's support multiple stable coin values, for eth also WBTC

**Review by gemcoder21 — APPROVED** [review]

File: crates/gem_evm/src/uniswap/path.rs (hunk 2 above)
Comment (0xh3rman) [reply crates/gem_evm/src/uniswap/path.rs:82]: token_pairs could be only one element

File: crates/gem_evm/src/uniswap/path.rs (hunk 3 above)
Comment (0xh3rman) [reply crates/gem_evm/src/uniswap/path.rs:53]: added, will modify the request part later

**Pull request merged**